core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -401,6 +402,30 @@ public static long getMs(PropertyKey key) {
   }
 
   /**
+   * Gets a set of properties that share a given common prefix key as a map. E.g., if A.B=V1 and
+   * A.C=V2, calling this method with prefixKey=A returns a map of {B=V1, C=V2}, where B and C are
+   * also valid properties. If no property shares the prefix, an empty map is returned.
+   *
+   * @param prefixKey the prefix key
+   * @return a map of trailing properties aggregated by the prefix
+   */
+  public static Map<PropertyKey, String> getTrailingProperties(PropertyKey prefixKey) {
+    int prefixKeyLen = prefixKey.toString().length();
+    Map<PropertyKey, String> ret = Maps.newHashMap();
+    for (Map.Entry<String, String> entry: PROPERTIES.entrySet()) {
+      String key = entry.getKey();
+      if (key.length() > prefixKeyLen + 1 && key.startsWith(prefixKey.toString()) &&
+          key.charAt(prefixKeyLen) == '.') {
+        String trailingKey = key.substring(prefixKeyLen + 1);
```

Comment:
Similarly, do you think it makes sense to move this logic to `PropertyKey`, ie. `isNestedKey`?